tutorials/suspending-operations.html

```diff
@@ -0,0 +1,348 @@
+      <li><a href="#page-suspend-api">API methods</a></li>
+      <li><a href="#page-suspend-demo">Supsend - an example</a></li>
+    </ul>
+  </div>
+
+  <div class="example-container head-gap clearfix" name="suspend-feature">
+    <h3 id="page-suspend-feature">Suspending operations</h3>
+    <p>Suspending (also called "batching") operations might be useful if you want to work on improving the performance.
+      Batching several costly operations can decrease the number of reflows and render cycles for CRUD operations.
+      It results in less layout trashing, fewer freezes, and a more responsive feel.
+      <br>
+      There are several <strong>API methods</strong> which allow suspending:
+    </p>
+
+    <ul>
+      <li><code>suspendRender</code> and <code>resumeRender</code></li>
+      <li><code>batchRender</code></li>
+      <li><code>suspendExecution</code> and <code>resumeExecution</code></li>
+      <li><code>batchExecution</code></li>
+      <li><code>batch</code></li>
+    </ul>
+
+    <p>
+    By using these methods you can suspend:
+    <ul>
+      <li>rendering</li>
+      <li>execution</li>
+      <li>both rendering and the execution.</li>
+    </ul>
+
+    <br>
+      Term "rendering" refers directly to DOM rendering, the term
+      "execution" refers to all operations that are different from DOM rendering.
+      Currently only the indexes recalculation allows to postpone the process.
+    </p>
+    <p>
+      Note that this is an advanced feature which can improve the performance but should be used with caution.
+      Misuse can result in some flickering, glitches, or other visual distortion.
+    </p>
+
+  <div class="example-container head-gap clearfix" name="suspend-api">
+    <h3 id="page-suspend-api">API methods</h3>
+    <h4>Suspend rendering</h4>
+
+    <p>
+      <strong>suspendRender</strong> and <strong>resumeRender</strong>
+      <br>
+      To suspend the rendering process you can use <code>suspendRender</code> method just before the actions you want to batch.
+      This is a manual approach - after suspending you have to remember to resume the process with the <code>resumeRender</code> method.
+      <pre>
+        <code class="lang-js">
+hot.suspendRender(); // suspend rendering
+hot.alter('insert_row', 5, 45);
+hot.setDataAtCell(1, 1, 'x');
+hot.resumeRender(); // remember to resume rendering
+        </code>
+      </pre>
+
+      <p><strong>batchRender</strong> </p>
+      The <code>batchRender</code> method is less prone to human errors since the operations will be calculated after executing the callback.
+      You don't have to remember about resuming the operations.
+      However, there is a drawback to this method: it doesn't support async operations.
+      <pre>
+        <code class="lang-js">
+hot.batchRender(() => {
+hot.alter('insert_row', 5, 45);
+hot.setDataAtCell(1, 1, 'x');
+// The table will be rendered once after executing the callback
```

Comment:
Content of the closure should have indentation?
```suggestion
  hot.alter('insert_row', 5, 45);
  hot.setDataAtCell(1, 1, 'x');
  // The table will be rendered once after executing the callback
```

Reply:
I think it will stand out slightly, for instance, in the migration guide we have:
![image](https://user-images.githubusercontent.com/20072336/103340184-c1f28600-4a83-11eb-890f-09b439e02406.png)

and if I make an indent here:

![image](https://user-images.githubusercontent.com/20072336/103340245-e0f11800-4a83-11eb-996a-fc9ea64bf8d2.png)

Reply:
But you have `batchRender` here with a closure, so it will look more like:

```
hot.batchRender(() => {
    hot.alter('insert_row', 5, 45);
    hot.setDataAtCell(1, 1, 'x');
```

Reply:
Ah yes, maybe it's good to read from the top to bottom @scarletfog!

I will correct that...